Automattic/wp-calypso
# State: Implement currentUser settings subtree

Closes #7180 

_Note: This pull request is a work-in-progress. Most of the basic logic has been implemented, but it is lacking tests, comments, state deduplication between preferences and current user subtrees, and schema consideration for persistence._

This pull request seeks to generalize the preferences state into a general purpose current user settings retrieval and update subtree.

__Testing instructions:__

There should be no changes to the application. Verify that no regressions occur, particularly in using preferences state (first view, editor media gallery help).

Ensure that Mocha tests pass:

```
npm run test-client
```

Test live: https://calypso.live/?branch=add/state-current-user-settings
